src/logic/map_objects/description_manager.cc

```diff
@@ -78,8 +80,25 @@ void DescriptionManager::register_directory(const std::string& dirname,
 						register_description(
 						   object_name, FileSystem::fs_dirname(file) + "init.lua", attributes, caller);
 						register_attributes(attributes, object_name);
+						all_registered_items.insert(object_name);
 					}
 				}
+				if (all_registered_items.empty()) {
+					log_warn("Registry file %s does not define any units", file.c_str());
+				} else if ((g_verbose || caller.first != RegistryCallerType::kDefault) &&
+				           all_registered_items.size() > 1) {
+					std::string all;
+					for (const std::string& str : all_registered_items) {
```

Comment:
This result in a lot of memory allocations, is there a better way? (e.g. reserving enough space in a buffer)

Reply:
You could use `all.resize(x)` with a guessed value for x (~`10 * all_registered_items.size()`) beforehand to reduce the number of necessary string resizing.